### PR TITLE
Adds 3DTILES_content_gltf extension

### PR DESCRIPTION
**Pull request by lilleyse**
Fixes https://github.com/CesiumGS/3d-tiles/issues/385

Direct Link|Draft Version|Changes
--|--|--
[`3DTILES_content_gltf`](https://github.com/CesiumGS/3d-tiles/blob/3DTILES_content_gltf/extensions/3DTILES_content_gltf) | 0.0.0 | Initial draft

This extension allows a tileset to use [glTF 2.0](https://github.com/KhronosGroup/glTF/tree/master/specification/2.0) and [GLB](https://github.com/KhronosGroup/glTF/tree/master/specification/2.0#glb-file-format-specification) models directly as tile content.

Using glTF directly simplifies runtime implementations and content pipelines that already support glTF but don't support 3D Tiles native formats. glTF models may be extended with instancing, feature metadata, and compression extensions to achieve near feature parity with the existing 3D Tiles formats.

The [example tileset](https://github.com/CesiumGS/3d-tiles/tree/3DTILES_content_gltf/extensions/3DTILES_content_gltf/examples/tileset) can be loaded on the [feature-metadata](https://github.com/CesiumGS/cesium/tree/feature-metadata) branch in CesiumJS.

### CODE REVIEW

**Comment by lilleyse** [comment]
## Format Comparison

The sections below describe the differences between native 3D Tiles formats and glTF.

### Batched 3D Model (b3dm)

Since [`b3dm`](../../specification/TileFormats/Batched3DModel/README.md) is a thin wrapper around binary glTF there aren't too many differences when using glTF directly.

* `RTC_CENTER` may be represented as a glTF node translation.
* Batch Table and Batch IDs may be stored with [`EXT_feature_metadata`](https://github.com/CesiumGS/glTF/pull/1).

### Instanced 3D Model (i3dm)

[`i3dm`](../../specification/TileFormats/Instanced3DModel) instances a glTF model (embedded or external) and provides per-instance transforms and feature IDs.

* [`EXT_mesh_gpu_instancing`](https://github.com/KhronosGroup/glTF/tree/master/extensions/2.0/Vendor/EXT_mesh_gpu_instancing) may be used to instance glTF nodes.
* `RTC_CENTER` may be represented as a glTF node translation.
* Batch Table and Batch IDs may be stored with [`EXT_feature_metadata`](https://github.com/CesiumGS/glTF/pull/1). Per-instance properties are supported.
* No glTF support for compressed per-instance translations and rotations, uniform scale, multi-instance properties, and `EAST_NORTH_UP`.
* Longer term a 3D Tiles vector tile format may instance models as point features.

### Point Cloud (pnts)

glTF natively supports point clouds with the primitive mode `0` (`POINT`). Concepts in the [`pnts`](../../specification/TileFormats/PointCloud) format map well to glTF.

* Feature table properties like like `POSITION`, `COLOR`, and `NORMAL` may be stored as glTF attributes.
* [`EXT_meshopt_compression`](https://github.com/KhronosGroup/glTF/pull/1830) and [`KHR_mesh_quantization`](https://github.com/KhronosGroup/glTF/tree/master/extensions/2.0/Khronos/KHR_mesh_quantization) may be used for point cloud compression. `POSITION_QUANTIZED`, `NORMAL_OCT16P`, `RGB565`, and [`3DTILES_draco_point_compression`](https://github.com/CesiumGS/3d-tiles/tree/master/extensions/3DTILES_draco_point_compression) are not directly supported in glTF.
* `RTC_CENTER` may be represented as a glTF node translation.
* Batch Table and Batch IDs may be stored with [`EXT_feature_metadata`](https://github.com/CesiumGS/glTF/pull/1). The extension supports both per-point properties and multi-point properties in the same glTF.
* No glTF support for `CONSTANT_RGBA`.

### Composite (cmpt)

All inner contents may be bundled into the same glTF as separate nodes, meshes, or primitives, up to the tileset author's discretion. [`EXT_feature_metadata`](https://github.com/CesiumGS/glTF/pull/1) supports multiple feature collections.

**Comment by pjcozzi** [comment]
@lilleyse please LMK when this is ready. for me to review.  Should anyone else review first?

**Comment by lilleyse** [comment]
A discussion about optional vs. required in a 3d-tiles-validator [PR](https://github.com/CesiumGS/3d-tiles-validator/pull/209#discussion_r511141144):


> [3DTILES_content_gltf](https://github.com/CesiumGS/3d-tiles/tree/3DTILES_content_gltf/extensions/3DTILES_content_gltf) says it needs to go in `extensionsRequired` but now I'm thinking about the interaction with [3DTILES_layers](https://github.com/CesiumGS/3d-tiles/pull/432) and how someone might create a tileset with both b3dm and glTF layers so that the b3dm layer can be a fallback if the client doesn't support `3DTILES_content_gltf`. That's a possible use case for allowing `3DTILES_content_gltf` to be optional.

**Review by sanjeetsuhag — COMMENTED** [review]
Spec looks good, but I noticed some problems with the example in the `3d-tiles-next-experimental` branch.

File: extensions/3DTILES_content_gltf/schema/tileset.3DTILES_content_gltf.schema.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        "extensionsRequired": {
+            "type": "array",
+            "description": "An array of glTF extensions required by glTF content in the tileset.",
+            "items": {
+                "type": ["string"]
+            },
+            "minItems": 1
+        },
+        "extensions": {
+            "$ref": "extension.schema.json"
```
Comment (sanjeetsuhag) [inline extensions/3DTILES_content_gltf/schema/tileset.3DTILES_content_gltf.schema.json:24]: This file does not exist.

File: extensions/3DTILES_content_gltf/schema/tileset.3DTILES_content_gltf.schema.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        },
+        "extras" : {
+            "$ref": "extras.schema.json"
```
Comment (sanjeetsuhag) [inline extensions/3DTILES_content_gltf/schema/tileset.3DTILES_content_gltf.schema.json:27]: This file does not exist.

File: extensions/3DTILES_content_gltf/README.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
+# 3DTILES_content_gltf
+
+## Contributors
+
+* Sean Lilley, Cesium
+
+## Status
+
+Draft
```
[the remark this answers is not in the document]
Comment (lilleyse) [reply extensions/3DTILES_content_gltf/README.md:9]: I think we should keep it until these extensions get merged into master

**Comment by sanjeetsuhag** [comment]
Looks good for `v.0.0.0`. Thanks @lilleyse

**Pull request merged**